ui/app/ducks/gas/gas.duck.js

```diff
@@ -161,6 +171,39 @@ async function fetchExternalBasicGasEstimates(dispatch) {
     setStorageItem('BASIC_PRICE_ESTIMATES_LAST_RETRIEVED', timeRetrieved),
   ]);
   dispatch(setBasicPriceEstimatesLastRetrieved(timeRetrieved));
+  return basicEstimates;
+}
+
+async function fetchEthGasPriceEstimates(state) {
+  const chainId = getCurrentChainId(state);
+  const timeLastRetrieved =
+    (await getStorageItem(`${chainId}_BASIC_PRICE_ESTIMATES_LAST_RETRIEVED`)) ||
+    0;
+
+  const cachedBasicEstimates = await getStorageItem(
+    `${chainId}_BASIC_PRICE_ESTIMATES`,
+  );
+  let basicEstimates = cachedBasicEstimates;
+  if (Date.now() - timeLastRetrieved > 75000 || !cachedBasicEstimates) {
+    const gasPrice = await global.eth.gasPrice();
+    const averageGasPriceInDecGWEI = getValueFromWeiHex({
+      value: gasPrice.toString(16),
+      numberOfDecimals: 4,
+      toDenomination: 'GWEI',
+    });
+    basicEstimates = {
+      average: Number(averageGasPriceInDecGWEI),
+    };
+  }
+  const timeRetrieved = Date.now();
+
+  await Promise.all([
```

Comment:
This step still seems to occur even when the estimate _hasn't_ changed. That could lead to the "last retrieved" time being updated over and over, preserving a stale estimate for longer than we intended.

One way to solve this would be to reverse the condition earlier on line 187 and return early.

e.g. something like

```
if (cachedBasicEstimates && Date.now() - timeLastRetrieved <= 75000) {
  return cachedBasicEstimates
}
```